share/spack/gitlab/cloud_pipelines/.gitlab-ci.yml

```
@@ -709,3 +709,36 @@ tutorial-protected-build:
   needs:
     - artifacts: True
       job: tutorial-protected-generate
+
+########################################
+# Machine Learning
+########################################
+.ml:
+  variables:
+    SPACK_CI_STACK_NAME: ml
+
+.ml-generate:
```

@adamjstewart My bad, I missed a line in my previous suggestions:
```suggestion
.ml-generate:
  extends: .ml
```